polaris-pipeline/Common/Clients/PdfGeneratorClient.cs

```diff
@@ -45,6 +42,16 @@ public async Task<Stream> ConvertToPdfAsync(Guid correlationId, string cmsAuthVa
             if (fileType == FileType.PDF)
                 return documentStream;
 
+            // For integration testing allow, PDF generation to be disabled
+            // By definition this only applies to non-PDFs
+            // In integration this will only be the HTML files generated by e.g. the PCD HTML files generated from case data
+            if (_configuration.IsConfigSettingEnabled(FeatureFlags.DisableConvertToPdfFeatureFlag))
+            {
+                // Use average time seen in production for PDF generation to simulate time typically taken
+                Thread.Sleep(500);
+                throw new InvalidOperationException($"{FeatureFlags.DisableConvertToPdfFeatureFlag} is configured");
```

Comment:
Does the thrown exception reflect that all documents in the test scenario are pdfs (as per the code comment I guess)? Will that be different to prod because activities will never take a finite amount of time as they do in prod? If there is a ceiling to the number of concurrent activities then will the test as is artificially keep us away from that ceiling?

Reply:
Yes - they are all PDFs except the PCD, so it excludes these other docs, so it gives a predictable document count, so if case has 100 PDFs we will have 100 documents.

Reply:
As discussed added a 500ms delay